src/test/groovy/com/github/rodionovsasha/commentservice/services/TopicServiceTest.groovy

```diff
@@ -0,0 +1,181 @@
+        e.message == "The user with id '2' is not active"
+    }
+
+    def "start throws when user not found"() {
+        when:
+        topicService.start("D'oh!", 999)
+
+        then:
+        def e = thrown(UserNotFoundException)
+        e.message == "The user with id '999' could not be found"
+    }
+
+    def "archive makes a topic archived"() {
+        given:
+        def topicId = 1
+        !topicService.getById(topicId).archived
+
+        when:
+        topicService.archive(topicId, 1)
+
+        then:
+        topicService.getById(topicId).archived
+    }
+
+    def "archive throws when user is inactive"() {
+        given:
+        def topicId = 1
+        !topicService.getById(topicId).archived
+
+        when:
+        topicService.archive(topicId, 2)
+
+        then:
+        !topicService.getById(topicId).archived
+        def e = thrown(InactiveUserException)
+        e.message == "The user with id '2' is not active"
+    }
+
+    def "archive throws when user not found"() {
+        given:
+        def topicId = 1
+        !topicService.getById(topicId).archived
+
+        when:
+        topicService.archive(topicId, 999)
+
+        then:
+        !topicService.getById(topicId).archived
+        def e = thrown(UserNotFoundException)
+        e.message == "The user with id '999' could not be found"
+    }
+
+    def "archive throws when user is not topic owner"() {
+        given:
+        def topicId = 3
+        !topicService.getById(topicId).archived
+
+        when:
+        topicService.archive(topicId, 1)
+
+        then:
+        !topicService.getById(topicId).archived
+        def e = thrown(TopicAccessException)
+        e.message == "Non-topic owner with id '1' is trying to archive the topic"
+    }
+
+    def "listForUser returns all topics for user ASC sorted"() {
+        when:
+        def topics = topicService.listForUser(1, new Sort(Sort.Direction.ASC, "id"))
+
+        then:
+        topics.size() == 2
+        topics.get(0).id == 1
+        topics.get(1).id == 2
+    }
+
+    def "listForUser returns all topics for user DESC sorted"() {
+        when:
+        def topics = topicService.listForUser(1, new Sort(Sort.Direction.DESC, "id"))
+
+        then:
+        topics.size() == 2
+        topics.get(0).id == 2
+        topics.get(1).id == 1
+    }
+
+    def "listForUser throws when user is inactive"() {
+        when:
+        topicService.listForUser(2, null)
+
+        then:
+        def e = thrown(InactiveUserException)
+        e.message == "The user with id '2' is not active"
+    }
+
+    def "listForUser throws when user not found"() {
+        when:
+        topicService.listForUser(999, null)
+
+        then:
+        def e = thrown(UserNotFoundException)
+        e.message == "The user with id '999' could not be found"
+    }
+
+    def "search returns topics with a fragment in title"() {
+        when:
+        def topics = topicService.search("e", 10)
```

Comment:
This test is a bit strange because it returns 2 out of 2 topic. Probably ideally it should return 2 out of 5 topics so we don't have an impression that it just returns everything

Reply:
Also a case is needed when nothing was found

Reply:
Results order should be checked in a separate test method probably not to mix things up